internal/service/service_component/datasource_service_component.go

```diff
@@ -135,9 +208,27 @@ func datasourceServiceComponentRead(_ context.Context, d *schema.ResourceData, m
 				}
 			}
 
-			// check optional kafka_authentication_method search criteria, if not set by a
-			// user match entries without kafka_authentication_method
-			if method, ok := d.GetOk("kafka_authentication_method"); ok {
+			filteredResult = append(filteredResult, c)
+		}
+
+		if len(filteredResult) == 0 {
+			var err_msg string
```

Comment:
Would you mind changing `err_msg` to camel Case for consistency?

Reply:
ah, yeah of course, it's python habit :sweat:

Reply:
np :pray: